File: 30-day-leetcoding-challenge/Construct Binary Search Tree from Preorder Traversal.java
class Solution {
    private void insert(TreeNode parent,TreeNode node)
    {
        if(node.val < parent.val)
        {
            if(parent.left == null)
                parent.left = node;
            else
                insert(parent.left,node);
        }else{
            if(parent.right == null)
                parent.right = node;
            else
                insert(parent.right,node);
        }
    }
    public TreeNode bstFromPreorder(int[] preorder) {
        if(preorder.length == 0)return null;
        TreeNode root = new TreeNode(preorder[0]);
        for(int i=1;i<preorder.length;i++)
            insert(root,new TreeNode(preorder[i]));
        return root;
    }
}

File: May LeetCoding Challenge/Construct Binary Search Tree from Preorder Traversal.cs
public class Solution {
    private TreeNode root;
    private void Insert(int val)
    {
        if(root == null)
        {
            root = new TreeNode(val);
            return;
        }
        TreeNode it = root;
        while(it != null)
        {
            if(val < it.val){
                if(it.left == null)
                {
                    it.left = new TreeNode(val);
                    break;
                }
                else
                    it = it.left;
            }else{
                if(it.right == null)
                {
                    it.right = new TreeNode(val);
                    break;
                }
                else
                    it = it.right;
            }
        }
    }
    public TreeNode BstFromPreorder(int[] preorder) {
        foreach(int val in preorder)Insert(val);
        return root;
    }
}
